diff --git a/net/acme-acmesh/Makefile b/net/acme-acmesh/Makefile
index 16d34fad6..04bf2ac8b 100644
--- a/net/acme-acmesh/Makefile
+++ b/net/acme-acmesh/Makefile
@@ -25,7 +25,7 @@ include $(INCLUDE_DIR)/package.mk
 define Package/acme-acmesh
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:=+acme-common +wget-ssl +ca-bundle +openssl-util +socat
+  DEPENDS:=+acme-common +curl +ca-bundle +openssl-util +socat
   TITLE:=ACME client acme.sh wrapper script
   URL:=https://acme.sh
   PKGARCH:=all
