diff --git a/net/shadowsocks-libev/files/shadowsocks-libev.init b/net/shadowsocks-libev/files/shadowsocks-libev.init
index 79ec2e8b3..0d1abc82f 100644
--- a/net/shadowsocks-libev/files/shadowsocks-libev.init
+++ b/net/shadowsocks-libev/files/shadowsocks-libev.init
@@ -27,7 +27,14 @@ ss_mkjson_server_conf() {
 }
 
 ss_mkjson_server_conf_() {
+	local dns nameserver_list
+
 	[ -n "$server_port" ] || return 1
+
+	for dns in $nameserver; do
+		append nameserver_list "$dns" ","
+	done
+
 	[ -z "$server" ] || json_add_string server "$server"
 	json_add_int server_port "$server_port"
 	[ -z "$method" ] || json_add_string method "$method"
@@ -35,6 +42,7 @@ ss_mkjson_server_conf_() {
 	[ -z "$password" ] || json_add_string password "$password"
 	[ -z "$plugin" ] || json_add_string plugin "$plugin"
 	[ -z "$plugin_opts" ] || json_add_string plugin_opts "$plugin_opts"
+	[ -z "$nameserver_list" ] || json_add_string nameserver "$nameserver_list"
 }
 
 ss_mkjson_ss_local_conf() {
@@ -307,7 +315,8 @@ validate_ss_server_section() {
 		'local_address:ipaddr' \
 		'local_ipv4_address:ip4addr' \
 		'local_ipv6_address:ip6addr' \
-		'bind_address:ipaddr'
+		'bind_address:ipaddr' \
+		'nameserver:list(ipaddr)'
 }
 
 validate_ss_tunnel_section() {
