diff --git a/utils/upx/Makefile b/utils/upx/Makefile
new file mode 100644
index 00000000..6f14cc59
--- /dev/null
+++ b/utils/upx/Makefile
@@ -0,0 +1,53 @@
+#
+# Copyright (C) 2011-2013 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=upx
+PKG_VERSION:=3.95
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-src.tar.xz
+PKG_SOURCE_URL:=https://github.com/upx/upx/releases/download/v$(PKG_VERSION)
+PKG_HASH:=3b0f55468d285c760fcf5ea865a070b27696393002712054c69ff40d8f7f5592
+
+HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/$(PKG_NAME)-$(PKG_VERSION)-src
+HOST_BUILD_DEPENDS:=ucl/host
+PKG_HOST_ONLY:=1
+
+include $(INCLUDE_DIR)/host-build.mk
+include $(INCLUDE_DIR)/package.mk
+
+define Package/upx
+  SECTION:=utils
+  CATEGORY:=Utilities
+  SUBMENU:=Compression
+  TITLE:=UPX the Ultimate Packer for eXecutables
+  URL:=https://upx.github.io/
+  BUILDONLY:=1
+endef
+
+define Package/upx/description
+  UPX is a free, portable, extendable, high-performance executable 
+  packer for several executable formats.
+endef
+
+define Host/Compile
+	rm -f $(HOST_BUILD_DIR)/src/.depend
+	$(MAKE) UPX_UCLDIR=$(STAGING_DIR_HOST) -C $(HOST_BUILD_DIR)/src \
+		CXXFLAGS_WERROR="" LDFLAGS="$(HOST_LDFLAGS) -static" \
+		CXX="$(HOSTCXX)"
+endef
+
+define Host/Install
+	$(CP) $(HOST_BUILD_DIR)/src/upx.out $(STAGING_DIR_HOST)/bin/upx
+endef
+
+define Host/Clean
+	rm -f $(STAGING_DIR_HOST)/bin/upx
+endef
+
+$(eval $(call HostBuild))
+$(eval $(call BuildPackage,upx))
diff --git a/utils/upx/patches/100-remove-check_whitespace.patch b/utils/upx/patches/100-remove-check_whitespace.patch
new file mode 100644
index 00000000..09247f25
--- /dev/null
+++ b/utils/upx/patches/100-remove-check_whitespace.patch
@@ -0,0 +1,14 @@
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -105,9 +105,9 @@
+ 
+ CHECK_WHITESPACE =
+ ifeq ($(shell uname),Linux)
+-CHECK_WHITESPACE = $(top_srcdir)/src/stub/scripts/check_whitespace.sh $(top_srcdir)
++CHECK_WHITESPACE =
+ ifneq ($(wildcard $(top_srcdir)/.git/.),)
+-CHECK_WHITESPACE = $(top_srcdir)/src/stub/scripts/check_whitespace_git.sh $(top_srcdir)
++CHECK_WHITESPACE =
+ endif
+ check-whitespace : ; $(CHECK_WHITESPACE)
+ endif
