diff --git a/net/v2ray/Makefile b/net/v2ray/Makefile
new file mode 100644
index 000000000..2acff7c24
--- /dev/null
+++ b/net/v2ray/Makefile
@@ -0,0 +1,62 @@
+#
+# Copyright (C) 2018 Keith Wong <keithwky@gmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_MAINTAINER:=Keith Wong <keithwky@gmail.com>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_NAME:=v2ray
+PKG_VERSION:=4.31.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/v2ray/v2ray-core/tar.gz/v${PKG_VERSION}?
+PKG_SOURCE_DATE:=20201011
+PKG_HASH:=4e475c700863320b92689cb36ef993133e9c799375bbfc4a2811da2283c3673f
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-core-$(PKG_VERSION)
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+
+GO_PKG:=v2ray.com/core
+GO_PKG_BUILD_PKG:=v2ray.com/core/main
+
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
+
+define Package/v2ray
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=A unified platform for anti-censorship
+  URL:=https://www.v2ray.com
+  DEPENDS:=$(GO_ARCH_DEPENDS)
+  PKGARCH:=all
+endef
+
+define Package/v2ray/description
+Project V is a set of network tools that help you to build your own
+computer network. It secures your network connections and thus protects
+your privacy. 
+endef
+
+define Build/Configure
+	# ensure that version and build info reflect PKG_VERSION and PKG_SOURCE_DATE of Makefile
+	$(SED) 's,\(version *= *\).*,\1\"$(PKG_VERSION)\",g' $(PKG_BUILD_DIR)/core.go
+	$(SED) 's,\(build *= *\).*,\1\"$(PKG_SOURCE_DATE)\",g' $(PKG_BUILD_DIR)/core.go
+
+	$(call GoPackage/Build/Configure)
+endef
+
+define Package/v2ray/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(GO_PKG_BUILD_BIN_DIR)/main $(1)/usr/bin/v2ray
+endef
+
+$(eval $(call GoBinPackage,v2ray))
+$(eval $(call BuildPackage,v2ray))
