diff --git a/feeds.conf.default b/feeds.conf.default
index 040dbdc551..48ffde5afc 100644
--- a/feeds.conf.default
+++ b/feeds.conf.default
@@ -2,6 +2,7 @@ src-git packages https://git.openwrt.org/feed/packages.git
 src-git-full luci https://git.openwrt.org/project/luci.git
 src-git-full routing https://git.openwrt.org/feed/routing.git
 src-git-full telephony https://git.openwrt.org/feed/telephony.git
+src-git-full waller https://github.com/yousong/waller.git
 #src-git-full video https://github.com/openwrt/video.git
 #src-git-full targets https://github.com/openwrt/targets.git
 #src-git-full oldpackages http://git.openwrt.org/packages.git
