diff --git a/files/etc/hotplug.d/ntp/20-ntpd-logger b/files/etc/hotplug.d/ntp/20-ntpd-logger
new file mode 100644
index 0000000000..7f1ff28f8b
--- /dev/null
+++ b/files/etc/hotplug.d/ntp/20-ntpd-logger
@@ -0,0 +1,4 @@
+#!/bin/sh
+[ $ACTION = "step" ]    && logger -t ntpd Time set, stratum=$stratum interval=$poll_interval offset=$offset
+[ $ACTION = "stratum" ] && logger -t ntpd Stratum change, stratum=$stratum interval=$poll_interval offset=$offset
+
