diff --git a/package/network/services/uhttpd/files/uhttpd.init b/package/network/services/uhttpd/files/uhttpd.init
index 6322473b97..8fe256cb55 100755
--- a/package/network/services/uhttpd/files/uhttpd.init
+++ b/package/network/services/uhttpd/files/uhttpd.init
@@ -4,6 +4,7 @@
 START=50
 
 USE_PROCD=1
+NICEPRIO=19
 
 UHTTPD_BIN="/usr/sbin/uhttpd"
 PX5G_BIN="/usr/sbin/px5g"
@@ -121,6 +121,7 @@ start_instance()
 	procd_set_param respawn
 	procd_set_param stderr 1
 	procd_set_param command "$UHTTPD_BIN" -f
+	procd_set_param nice "$NICEPRIO"
 
 	config_get config "$cfg" config
 	if [ -z "$config" ]; then
